adventurine.rb

```diff
@@ -1,26 +1,87 @@
 require 'pp' #so the board prints prettier
 
 board = [
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
-  ["_","_","_","_","_","_",],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"],
+  ["_","_","_","_","_","_"]
 ]
 
-user_pos = [0,0]
-a = [-1,0]
-s = [0,1]
-w = [0,-1]
-d = [1,0]
+user_x = 0 
+user_y = 0
+valid_moves = ['w', 'a', 's', 'd', 'x', 'p']
 
-pp board
-print "You are in the northwestern most corner of a dungeon. Use 'a' to move west, 's' to move south, 'w' to move north, and 'd' to move east: "
-user_input = gets.chomp.downcase
+puts "What is your character's name?"
+character = gets.chomp
 
-if user_input == 'a' 
-  user_pos = user_pos.zip(a).map { |x, y| x + y } #if input is a then add user_pos array to a array and store it in user_pos
-elsif user_input == 'b' 
-  user_pos = user_pos.zip(b).map { |x, y| x + y }
-end #same thing if input is b...etc
+board[user_x][user_y] = character.chars[0].upcase
+
+
+board.each do |row|
+  row.each do |column|
+    print "#{column} "
+  end
+  puts ""
+end
+puts "You are in the northwestern most corner of a dungeon."
+puts "Use 'a' to move west, 's' to move south, 'w' to move north, and 'd' to move east"
+puts "Typing 'p' will show your position and 'x' will quit"
+
+loop do
+  puts "WASD?"
+  move = gets.chomp.downcase
+  if valid_moves.include? move
+    if move == 'w' 
+      puts "Walking up..."
+      if user_y - 1 >= 0
+        board[user_y][user_x] = "_"
+        user_y -= 1
+        board[user_y][user_x] = character.chars[0].upcase
+      else
+        puts "You bumped into a wall"
+      end
+    elsif move == 'a' 
+      puts "Walking left..."
+      if user_x - 1 >= 0
+        board[user_y][user_x] = "_"
+        user_x -= 1
+        board[user_y][user_x] = character.chars[0].upcase
+      else
+        puts "You bumped into a wall"
+      end
+    elsif move == 's' 
+      puts "Walking down..."
+      if user_y + 1 < board.length
+        board[user_y][user_x] = "_"
+        user_y += 1
+        board[user_y][user_x] = character.chars[0].upcase
+      else
+        puts "You bumped into a wall"
+      end
+    elsif move == 'd' 
+      puts "Walking right..."
+      if user_x + 1 < board[0].length
+        board[user_y][user_x] = "_"
+        user_x += 1
+        board[user_y][user_x] = character.chars[0].upcase
+      else
+        puts "You bumped into a wall"
+      end
+    elsif move == 'p'
+      puts "You're at: #{user_x}, #{user_y}"
```

Comment:
This was to mainly help me debug. I wanted to be sure that the position was being increased or decreased properly